apps/app/src/components/CreateKeyModal/CreateKeyModal.tsx

```diff
@@ -171,46 +171,40 @@ export const CreateKeyModal: FC<{ onClose: () => void }> = ({ onClose }) => {
-                        >
-                            Name
-                        </label>
-                        <div className="flex items-center gap-2 text-neutral-500">
-                            <input
+                        <div className={titleClass}>
+                            <TextField
+                                label="Name"
                                 type="text"
                                 id="name"
-                                className={cx(
-                                    'text-sm rounded-lg block w-full p-2.5 border',
-                                    errors.name
-                                        ? 'bg-red-900 bg-opacity-20 border-red-500 focus-visible:outine-red-500'
-                                        : 'focus:ring-blue-500 focus:border-blue-500 bg-neutral-50 border-neutral-300 dark:bg-neutral-600 dark:border-neutral-500 dark:placeholder-neutral-400 dark:text-white'
-                                )}
+                                errorMessage={
+                                    errors.name == undefined
+                                        ? undefined
+                                        : 'Invalid name'
+                                }
                                 placeholder="My Awesome Key"
-                                required
-                                {...register('name')}
+                                register={register}
                             />
                         </div>
                     </div>
                     <div>
-                        <span className="block text-base font-medium text-neutral-900 dark:text-neutral-300 mb-2">
-                            Key Permissions
-                        </span>
+                        <div className={titleClass}>Key Permissions</div>
                         <div className="grid items-center gap-2 text-neutral-500">
                             <Checkbox
                                 id="Full Access"
-                                label="Full Access"
+                                // {...register('Full Access')}
                                 register={register}
-                            />
+                            >
+                                Full Access
+                            </Checkbox>
                             {Object.keys(Perms).map((perm) => (
                                 <Checkbox
                                     id={perm}
                                     key={perm}
-                                    label={perm}
                                     register={register}
-                                    disabled={watch('Full Access')}
-                                />
+                                    isDisabled={watch('Full Access')}
+                                >
+                                    {perm}
```

Comment:
perm?

Reply:
Supposedly permissions. Just moved it from the label parameter to children.